cypress/integration/invoices/invoices.createCreditInvoice.spec.js

```diff
@@ -0,0 +1,71 @@
+import TopMenu from '../../support/fragments/topMenu';
+import NewInvoice from '../../support/fragments/invoices/newInvoice';
+import NewInvoiceLine from '../../support/fragments/invoices/newInvoiceLine';
+import Invoices from '../../support/fragments/invoices/invoices';
+import TestType from '../../support/dictionary/testTypes';
+import VendorAddress from '../../support/fragments/invoices/vendorAddress';
+import NewFund from '../../support/fragments/finance/funds/newFund';
+import Funds from '../../support/fragments/finance/funds/funds';
+import { getCurrentFiscalYearCode } from '../../support/utils/dateTools';
+import Helper from '../../support/fragments/finance/financeHelper';
+import Transaction from '../../support/fragments/finance/fabrics/newTransaction';
+
+describe('ui-invoices: Credit Invoice creation', () => {
+  const invoice = { ...NewInvoice.defaultUiInvoice };
+  const vendorPrimaryAddress = { ...VendorAddress.vendorAddress };
+  const invoiceLine = { ...NewInvoiceLine.defaultUiInvoiceLine };
+  const fund = { ...NewFund.defaultFund };
+  const subtotalValue = 100;
+
+  before(() => {
+    cy.getToken(Cypress.env('diku_login'), Cypress.env('diku_password'));
+    cy.getOrganizationApi({ query: `name=${invoice.vendorName}` })
+      .then(({ body }) => {
+        invoice.accountingCode = body.organizations[0].erpCode;
+        Object.assign(vendorPrimaryAddress,
+          body.organizations[0].addresses.find(address => address.isPrimary === true));
+      });
+    cy.getBatchGroups()
+      .then(({ body }) => {
+        invoice.batchGroup = body.batchGroups[0].name;
+      });
+    Funds.createFundViaUI(fund)
+      .then(
+        () => {
+          Funds.addBudget(100);
+          Funds.checkCreatedBudget(fund.code, getCurrentFiscalYearCode());
+        }
+      );
+    invoiceLine.subTotal = -subtotalValue;
+    cy.visit(TopMenu.invoicesPath);
+  });
+
+  it('C343209 Create a credit invoice', { tags: [TestType.smoke] }, () => {
+    Invoices.createDefaultInvoiceViaUi(invoice, vendorPrimaryAddress);
+    Invoices.createInvoiceLine(invoiceLine, false);
+    Invoices.addFundDistributionToLine(invoiceLine, fund);
+    Invoices.approveInvoice();
+    // check transactions after approve
+    cy.visit(TopMenu.fundPath);
+    Helper.searchByName(fund.name);
+    Helper.selectFromResultsList();
+    Funds.openBudgetDetails(fund.code, getCurrentFiscalYearCode());
+    Funds.openTransactions();
+    const valueInTransactionTable = `$${subtotalValue.toFixed(2)}`;
+    const pendingPaymentTransaction = new Transaction('Pending payment', valueInTransactionTable, fund.code, '', 'Invoice', '');
```

Comment:
can't find quickly implementation of "A pending payment transaction is created that lowers the "Awaiting payment" value of the budget by 100". If it's presented may be it will be better to reorganize your code.

Reply:
It's in 'checkTransaction' step. However phrasing in test itself can be changed. I will update it.